### PR TITLE
Change >>_ from default response to stub response

### LINKED ISSUE #1035: What's the meaning of ">> _" in an interaction
The only reference I found is in the "valid but senseless" example.
I would have thought it replaces "something" from the mocked method.
But at least for a `Spy` it does call the real method instead of returning "something".
Does that mean "return the default value"? Which for a spy would of course be the result of the real method.
But what sense does it have then if it is equivalent to just omitting it?
At least it should be documented somewhere.

### PR DESCRIPTION
**Pull request by leonard84**
fixes #1035

This change is [Reviewable](https://reviewable.io/reviews/spockframework/spock/1090)

### CODE REVIEW

File: docs/interaction_based_testing.adoc [hunk 1]
```diff
@@ -681,6 +681,18 @@ subscriber.receive(_) >>> ["ok", "fail", "ok"] >> { throw new InternalError() }
 This will return `"ok", "fail", "ok"` for the first three invocations, throw `InternalError`
 for the fourth invocations, and return `ok` for any further invocation.
 
+
+=== Returning a default response
+
+If you don't really care what you return, but you must return a non-null value, you can use `_`.
```
Comment (Vampire) [inline docs/interaction_based_testing.adoc]: It is not non-`null` in all cases, is it?
I mean to remember having seen in the code that under some circumstances it indeed returns `null`.

File: docs/interaction_based_testing.adoc [hunk 2]
```diff
@@ -681,6 +681,18 @@ subscriber.receive(_) >>> ["ok", "fail", "ok"] >> { throw new InternalError() }
...
+This will use the same logic to compute a response as `Stub` (see <<Stubs>>), so it is only really useful for `Mock` and `Spy` instances.
```
Comment (Vampire) [inline docs/interaction_based_testing.adoc]: As you found in the one test, there actually is a valid use-case for `Stub`s too, if you chain different responses and one of them should be the default one.
Would maybe make sense to mention this here and provide an example?

File: docs/release_notes.adoc [hunk 3]
```diff
@@ -3,6 +3,19 @@ include::include.adoc[]
 
 == 2.0-M3 (tbd)
 
+=== Breaking Changes
+
+==== New meaning of `>> _`
+The meaning of `>> \_` has changed form "use the default response" to "return a stubbed value" (<<interaction_based_testing.adoc#_returning_a_default_response,Docs>>).
+The original behavior was only ever documented in the Javadocs and was basically the same to just omitting it.
+With the new behavior, you can have a `Mock` or `Spy` return the same value as a `Stub`.
+This is also more in line with the fact that `_` in arguments constraints stands for any non-`null` value, while a `Mock` returns `null` by default.
```
Comment (Vampire) [inline docs/release_notes.adoc]: This is either wrong or the documentation has a bug.
Docs say:
> 1 * subscriber.receive(\_)              // any single argument (including null)
> 1 * subscriber.receive(*_)             // any argument list (including the empty argument list)
> 1 * subscriber.receive(!null)          // any non-null argument

File: docs/interaction_based_testing.adoc (hunk 1 above)
Comment (Vampire) [reply docs/interaction_based_testing.adoc]: Hm, maybe it was only for `void` and `Void` return types.

File: docs/release_notes.adoc (hunk 3 above)
Comment (leonard84) [reply docs/release_notes.adoc]: yeah its wrong

File: docs/interaction_based_testing.adoc (hunk 1 above)
Comment (leonard84) [reply docs/interaction_based_testing.adoc]: Yes `org.spockframework.mock.EmptyOrDummyResponse` only returns `null` for `void` methods.

**Pull request merged**